ipfssync/subpub/stream.go

```diff
@@ -0,0 +1,60 @@
+package subpub
+
+import (
+	"bufio"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/network"
+	libpeer "github.com/libp2p/go-libp2p-core/peer"
+	"go.vocdoni.io/dvote/log"
+)
+
+func (ps *SubPub) handleStream(stream network.Stream) {
+	// 'stream' will stay open until you close it (or the other side closes it).
+
+	peer := stream.Conn().RemotePeer()
+	log.Infof("connected to peer %s: %+v", peer, stream.Conn().RemoteMultiaddr())
+
+	// Any messages read from the stream are sent to the SubPub.Reader channel.
+	go ps.readHandler(stream)
+
+	// Create a buffer stream for non blocking write.
+	ps.Unicasts[peer] = bufio.NewWriter(stream)
```

Comment:
this might happen at any time, concurrently with calls to Unicast, so it seems racy.

Consider https://pkg.go.dev/sync#Map instead?

Reply:
thanks a lot for this! totally unaware. read the docs and tried implementing it, but got a bit stuck since  Load() returns an `interface{}` and ended up  falling back to a RWMutex https://github.com/vocdoni/vocdoni-node/pull/363/commits/11912aa200122d7b87572a3a5f020c91cd627375

Reply:
Ah, so the way you deal with that is:

```
var value valueType
m.Store(key, value)

[...]

value := m.Load(key).(valueType) // shouldn't panic unless you stored a different type
```

Reply:
marvellous! https://github.com/vocdoni/vocdoni-node/pull/363/commits/1317f49972a60d98227348eb7868e62b29f4e175